### DISCUSSION TITLE
maybe add PHP@7.4 back to homebrew as "disabled version" ?

### DISCUSSION THREAD

**Discussion by buildyuan** [Homebrew/discussions#5184] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.11
ORIGIN: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git
HEAD: d7d4c8266210c024b93a450a7d357cec0b46a1bb
Last commit: 4 days ago
Core tap JSON: 08 Mar 02:13 UTC
Core cask tap JSON: 08 Mar 02:13 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_API_DOMAIN: https://mirrors.tuna.tsinghua.edu.cn/homebrew-bottles/api
HOMEBREW_BOTTLE_DOMAIN: https://mirrors.tuna.tsinghua.edu.cn/homebrew-bottles
HOMEBREW_BREW_GIT_REMOTE: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/homebrew-core.git
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.3.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  consul
```

### Description of issue

I am currently doing both PHP and Golang development. 
Is it possible to bring back PHP 7.4  as "disabled version" ?
For example, the Homebrew has golang 1.16 (disabled) / php 8.0 (disabled). 

Because some projects  too old and not suitable for upgrading to PHP 8.

**Comment by ZhongRuoyu** [comment]
`php@7.4` was removed from `homebrew/core` in Homebrew/homebrew-core#153820 because it was no longer supported upstream and thus did not meet the [criteria for inclusion](https://docs.brew.sh/Versions#acceptable-versioned-formulae) anymore.

If you really need it despite its unsupported status, consider alternative installation methods or [hosting it in your own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap).

**Reply by buildyuan** [reply]
I understand the relevant rules,
but since the `homebrew/core` allows other unsupported upstream like  [go@1.16: deprecate](https://github.com/Homebrew/homebrew-core/pull/96941) /   [php@8.0: remove livecheck](https://github.com/Homebrew/homebrew-core/pull/156069), maybe `php@7.4` can also be added to the source.

**Reply by ZhongRuoyu** [reply, marked as answer]
We do *not* allow unsupported software. The two formulae you mentioned were deprecated when they became unsupported, and have already been disabled now, which means you can't install them anymore. They will eventually be removed according to https://docs.brew.sh/Deprecating-Disabling-and-Removing-Formulae#removal.

`php@7.4` has already went through the whole deprecate -> disable -> remove cycle. (Versioned formulae can be removed straight away but it did went through the deprecation cycle anyway.)
